lib/galaxy/tools/actions/__init__.py

```diff
@@ -630,10 +633,11 @@ def handle_output(name, output, hidden=None):
                         # Add elements to top-level collection, unless nested...
                         current_element_identifiers = element_identifiers
                         current_collection_type = output.structure.collection_type
+                        assert current_collection_type
+                        # TODO: replace following line with formal abstractions for doing this.
+                        current_collection_type = ":".join(current_collection_type.split(":")[1:])
 
                         for parent_id in output_part_def.parent_ids or []:
-                            # TODO: replace following line with formal abstractions for doing this.
-                            current_collection_type = ":".join(current_collection_type.split(":")[1:])
```

Comment:
@mvdbeek This is the source of the failing API test, opening a PR to revert this change.

Reply:
Ah, yes, that makes sense. Maybe a small function like `get_subcollection_type` would make this clearer ?